lib/zexbox/telemetry.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Zexbox.Telemetry do
```

Comment:
Is the purpose of this primarily for the types?

Reply:
No, it's purpose is two-fold:
1. The `attach` function was defined in both the `Zexbox.Metrics` and `Zexbox.Logging` modules so it was repeated code (and both were public functions so it was a bit confusing to have the same functionality defined twice). So this just dries things up a bit.
2. It provides an Elixir interface for the `:telemetry.attach/4` function since it's effectively part of the functionality that we want to provide (i.e. writing custom logs).